libsolidity/codegen/LValue.cpp

```diff
@@ -338,10 +338,35 @@ void StorageItem::storeValue(Type const& _sourceType, SourceLocation const& _loc
 		if (m_dataType->category() == Type::Category::Array)
 		{
 			m_context << Instruction::POP; // remove byte offset
-			ArrayUtils(m_context).copyArrayToStorage(
-				dynamic_cast<ArrayType const&>(*m_dataType),
-				dynamic_cast<ArrayType const&>(_sourceType)
-			);
+			auto const& sourceArrayType = dynamic_cast<ArrayType const&>(_sourceType);
+			if (
+				(sourceArrayType.baseType()->isDynamicallySized() || dynamic_cast<StructType const*>(sourceArrayType.baseType())) &&
```

Comment:
I left this PR to be only for `sol->yul ` implementation. It would be easier to merge. Than will follow up with things that can be enabled in olcodegen as well